warehouse/query-core/src/main/java/datawave/query/iterator/SourceManager.java

```diff
@@ -178,8 +210,16 @@ public void seek(Range range, Collection<ByteSequence> columnFamilies, boolean i
             
             try {
                 lastKey = child.getTopKey();
+                lastValue = child.getTopValue();
+                rangeDone = false;
```

Comment:
Lets move the setting of rangeDone to false to the top of the method.  This will allow us to remove a mess of sets below and guarentee we did not forget to set it on some path therein.

Reply:
Perhaps setting the lastKey and lastValue to null at the top would also be good